### ISSUE TITLE
some files ignored?

### ISSUE THREAD

**Issue by rhyek** [vuejs/eslint-plugin-vue#59]
**Tell us about your environment**

* **ESLint Version:** 4.1.1
* **eslint-plugin-vue Version:** 3.3.0
* **Node Version:** 8.1.2

**Please show your full configuration:**

```js
{
  "extends": [
    "eslint:recommended",
    "plugin:vue/recommended" // or 'plugin:vue/base'
  ],
  "rules": {
    // override/add rules' settings here
    "vue/html-no-self-closing": "error",
    "vue/require-v-for-key": "error"
  }
}

```

**What did you do? Please include the actual source code causing the issue.**

- [productos.vue](https://gist.github.com/rhyek/f3fe31be82ad539c4264204025f933c2): eslint does *not* work at all for `<template>` section, but does for `<script>`.
- [atributos.vue](https://gist.github.com/rhyek/c2703cedd7641589e3942abd1e7cd789): eslint works perfectly.

**What did you expect to happen?**

I am using the VS Code ESLint extension and I expect to see Vue template errors while typing.

**What actually happened? Please include the actual, raw output from ESLint.**

Some files display such errors, others don't even though ESLint does show errors for the `<script>` section.

Relevant: vuejs/vetur#235.

**Comment by michalsnik** [comment]
Hey @rhyek, thanks for posting this issue.

Please disable rule `vue/html-no-self-closing` and see if that helps.

This rule has been disabled in recommended configuration because of the problems it was causing and false assumptions on which it was created in the first place.

**Comment by rhyek** [comment]
It's only on for testing purposes and works fine in one of the files. Also, it isn't the only rule I'm testing, as you can see.

**Comment by michalsnik** [comment]
Yes, but I'm trying to find the reason behind why one of your components is being linted properly and other not. First guess is that `vue/html-no-self-closing` causes this weird behaviour somehow, that's why I'm asking you to remove this rule and see if everything else works :)

**Comment by rhyek** [comment]
I commented out `vue/html-no-self-closing`, reloaded the window (which I know isn't necessary), and I see the same. `atributos.vue` does not show error if I type something like `<img/>` and `productos.vue` still does not show error for all the `v-for` without `:key`.

**Comment by michalsnik** [comment]
Ok, thanks. Can you now please confirm whether there are the same results in VS Code as by running './node_modules/.bin/eslint . --ext .vue' ?

**Comment by rhyek** [comment]
I get the following:
<pre>
$ ./node_modules/.bin/eslint --ext .vue .
undefined == true
AssertionError [ERR_ASSERTION]: undefined == true
    at Object.hasAttribute (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/utils/index.js:104:5)
    at Linter.Program (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/rules/no-invalid-template-root.js:34:28)
    at emitOne (events.js:120:20)
    at Linter.emit (events.js:210:7)
    at NodeEventGenerator.applySelector (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/lib/util/node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/lib/util/node-event-generator.js:294:22
)
    at NodeEventGenerator.enterNode (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/lib/util/node-event-generator.js:308:14)
    at CodePathAnalyzer.enterNode (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:6
02:23)
    at Traverser.enter (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/lib/linter.js:922:36)
    at Traverser.__execute (/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/estraverse/estraverse.js:397:31)
</pre>

**Comment by rhyek** [comment]
I figured it out. I added the following to my `.eslintrc` file:
```js
  "parserOptions": {
    "ecmaVersion": 2017, // irrelevant
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  },
```
Without it:
<pre>
$ find . -name productos.vue | xargs node_modules/.bin/eslint --debug
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +12ms
  eslint:ignored-paths Looking for ignore file in /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster +4ms
  eslint:ignored-paths Could not find ignore file in cwd +1ms
  eslint:cli-engine Processing /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit/productos.vue +5ms
  eslint:cli-engine Linting /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit/productos.vue +4ms
  eslint:config Constructing config file hierarchy for /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit +1ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/.eslintrc +7ms
  eslint:config-file Loading config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/.eslintrc +1ms
  eslint:config-file Loading plugin:vue/recommended +160ms
  eslint:config-file Attempting to resolve eslint-plugin-vue +1ms
  eslint:config-file Loading JS config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/index.js +4ms
  eslint:config-file Loading /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/config/base.js +98ms
  eslint:config-file Loading JS config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/config/base.js +1ms
  eslint:config-file Loading /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/conf/eslint-recommended.js +4ms
  eslint:config-file Loading JS config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/conf/eslint-recommended.js +0ms
  eslint:config Using /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/.eslintrc +99ms
  eslint:config-ops Using config from partial cache +1ms
  eslint:config-ops Apply environment settings to config +3ms
  eslint:config-ops Creating config for environment browser +0ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:cli-engine Linting complete in: 596ms +199ms

/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit/productos.vue
  154:15  error  Parsing error: Unexpected token ..

✖ 1 problem (1 error, 0 warnings)
</pre>
That 1 error is from the `<script>` section.

With it:
<pre>
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +13ms
  eslint:ignored-paths Looking for ignore file in /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster +4ms
  eslint:ignored-paths Could not find ignore file in cwd +1ms
  eslint:cli-engine Processing /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit/productos.vue +4ms
  eslint:cli-engine Linting /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit/productos.vue +5ms
  eslint:config Constructing config file hierarchy for /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/.eslintrc +6ms
  eslint:config-file Loading config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/.eslintrc +1ms
  eslint:config-file Loading plugin:vue/recommended +164ms
  eslint:config-file Attempting to resolve eslint-plugin-vue +3ms
  eslint:config-file Loading JS config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/index.js +8ms
  eslint:config-file Loading /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/config/base.js +99ms
  eslint:config-file Loading JS config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint-plugin-vue/lib/config/base.js +0ms
  eslint:config-file Loading /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/conf/eslint-recommended.js +9ms
  eslint:config-file Loading JS config file: /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/node_modules/eslint/conf/eslint-recommended.js +1ms
  eslint:config Using /mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/.eslintrc +96ms
  eslint:config-ops Using config from partial cache +1ms
  eslint:config-ops Apply environment settings to config +4ms
  eslint:config-ops Creating config for environment browser +2ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:config-ops Creating config for environment jquery +0ms
  eslint:cli-engine Linting complete in: 828ms +417ms

/mnt/c/Users/Carlos/Dev/compuservice/web/salesmaster/resources/assets/js/modulos/inventario/ingresos/edit/productos.vue
    3:5   error  Self-closing should not be used                     vue/html-no-self-closing
   18:11  error  'v-for' directives require 'v-bind:key' directives  vue/require-v-for-key
   22:11  error  'v-for' directives require 'v-bind:key' directives  vue/require-v-for-key
   30:9   error  'v-for' directives require 'v-bind:key' directives  vue/require-v-for-key
   31:11  error  'v-for' directives require 'v-bind:key' directives  vue/require-v-for-key
   42:15  error  'v-for' directives require 'v-bind:key' directives  vue/require-v-for-key
   51:13  error  'v-for' directives require 'v-bind:key' directives  vue/require-v-for-key
  105:8   error  'Decimal' is defined but never used                 no-unused-vars
  211:13  error  'response' is assigned a value but never used       no-unused-vars

✖ 9 problems (9 errors, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
</pre>
And now I can see all these errors in the editor!

It feels to me this might be an issue with `eslint` directly?

**Comment by michalsnik** [comment]
Generally it's not an issue. Parser doesn't know how to handle object spread operator, which seems understandable. Too bad that editor didn't show this error, but it's an upstream issue if anything :) I will however add this setting to recommended configuration in order to prevent issues like this in the future. Thanks for investigating this!

**Comment by michalsnik** [comment]
I updated base config and released it in `v3.5.0`
